chapter05/ajd/exercise2.pl

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+
+if ( !scalar @ARGV ) {
+  die "usage: $0 <some data file>\n";
+}
+
+# Get the file read
+my %total_bytes;
+
+while ( <> ) {
+  if ( /^#/) { next };
+  my ( $source, $destination, $bytes ) = split;
+  $total_bytes{$source}{$destination} += $bytes;
+}
+
+# Sorting and printing
+foreach my $host ( sort { by_host_data( $a, $b, \%total_bytes ) } keys %total_bytes ) {
```

Comment:
This'll have to total the bytes for each host an unnecessary number of times to sort the hashes. Probably worth pre-computing the totals and sorting on that.